tools/configuration-converter.sh

```diff
@@ -0,0 +1,126 @@
+#!/bin/bash
+
+# Copyright Istio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script is used to convert istio configuration to sail operator configuration.
+# In the end of the execution new yaml file will be created with "sail-ISTIO_CONFIG_YAML" name.
+# Usage: ./configuration-converter.sh ISTIO_CONFIG_YAML_WITH_PATH, example:  ./configuration-converter.sh sample_config.yaml"
+set -e
+
+# Default values
+NAMESPACE="istio-system"
+VERSION=""
+INPUT=""
+OUTPUT=""
+
+# Function to show usage
+usage() {
+    echo "Usage: $0 <input> [output] [-n <namespace>] [-v <version>]"
+    echo "  <input>         : Input file (required)"
+    echo "  [output]        : Output file (optional, defaults to input's base directory with '-sail.yaml' suffix)"
+    echo "  -n <namespace>  : Namespace (optional, defaults to 'istio-system')"
+    echo "  -v <version>    : Istio Version (required)"
+    exit 1
+}
+
+# Ensure input file is provided
+if [[ -z "$1" ]]; then
+    echo "Error: Input file is required."
+    usage
+fi
+
+INPUT="$1"
+OUTPUT="$2"
+
+# Shift parameters if output is provided
+if [[ -n "$OUTPUT" && "$OUTPUT" != -* ]]; then
+    shift 2
+else
+    OUTPUT="$(dirname "$INPUT")/$(basename "$INPUT" .yaml)-sail.yaml"
+    shift 1
+fi
+
+# Ensure output is not a directory
+if [[ -d "$OUTPUT" ]]; then 
+    echo "Error: OUTPUT must be a file, not a directory."
+    exit 1
+fi
+
+# Parse optional namespace and version arguments
```

Comment:
How hard is it to modify this script to allow specifying the arguments in any order (e.g. `converter.sh -n namespace input-file.yaml -v version`)?